test/get-tracks.test.js

```diff
@@ -0,0 +1,62 @@
+const { getTracks } = require("..");
+
+test("getting data for empty url should return rejection", () => {
+  expect.assertions(1);
+  return expect(getTracks("")).rejects.toThrow();
+});
+
+test("getting data for non url string should return rejection", () => {
+  expect.assertions(1);
+  return expect(getTracks("arti39anptrackspotify:://https")).rejects.toThrow();
+});
+
+test("getting data for non spotify url string should return rejection", () => {
+  expect.assertions(1);
+  return expect(
+    getTracks("http://google.com/5a2w2tgpLwv26BYJf2qYwu")
+  ).rejects.toThrow();
+});
+
+test("get tracks for spotify track", () => {
+  expect.assertions(3);
+  return getTracks("https://open.spotify.com/track/5nTtCOCds6I0PHMNtqelas").then(
+    result => {
+      expect(Array.isArray(result)).toBe(true);
+      expect(result[0].name).toBe("Immaterial");
+      expect(result[0].external_urls.spotify).toContain("open.spotify.com/track");
+    }
+  );
+});
+
+test("get tracks for spotify artist", () => {
+  expect.assertions(3);
+  return getTracks("https://open.spotify.com/artist/5a2w2tgpLwv26BYJf2qYwu").then(
+    result => {
+      expect(Array.isArray(result)).toBe(true);
+      expect(result[0].name).toBe("Voices");
```

Comment:
I'm not sure, but this case gets the most popular tracks from the artist right?

If yes that will change over time and the test will fail, even though the code still works.

Instead you could check for artist instead of title on this one, which will hopefully still be the same.

Reply:
> I'm not sure, but this case gets the most popular tracks from the artist right?
> 
> If yes that will change over time and the test will fail, even though the code still works.
> 
> Instead you could check for artist instead of title on this one, which will hopefully still be the same.

oh i didnt know its the most popular i thought its all the songs the artist has but i edited it like you said